### DISCUSSION TITLE
Pixel cache does not seem to be garbage collected

### DISCUSSION THREAD

**Discussion by TobiasWolters** [dlemstra/Magick.NET#1647] [Help]
### Magick.NET version

Magick.NET-Q16-HDRI-AnyCPU, Version 13.6.0

### Environment (Operating system, version and so on)

Windows 11 Business, 23H2, 22631.3593

### Description

We are using Magick.NET to retrieve information about an image.

In particular, we are doing something along the lines of:

```C#
using var imageCollection = new MagickImageCollection();
magickImageCollection.Ping(imageStream);
...
```

We initialize Magick.NET in the following way:
```C#
var configurationFiles = ConfigurationFiles.Default;
configurationFiles.Policy.Data = """
                                         <policymap>
                                             <policy domain="resource" name="memory" value="96MiB"/>
                                             <policy domain="resource" name="map" value="512MiB"/>
                                             <policy domain="resource" name="disk" value="20GiB"/>
                                             <policy domain="system" name="max-memory-request" value="96MiB"/>
                                         </policymap>
                                         """;
MagickNET.Initialize(configurationFiles);
```

For large images, the entire pixel cache in RAM is used - i.e. all 512 MiB, allowed by the `map` parameter above.

However, the RAM allocated for the pixel cache never seems to be freed again, although the `MagickImageCollection` is disposed.

Is it possible to somehow ensure that the allocated RAM for the pixel cache is freed again?

### Steps to Reproduce

Use the setup from the description, providing a large image file via the `imageStream` parameter.

**Comment by dlemstra** [comment]
There is no pixel cache used when you `Ping` the image for most of the formats. The memory on the Native side is cleaned up with native code that calls free and returns memory to the operating system.  Memory allocated on the .NET side is not immediately returned to the operating system by the garbage collector so I suspect you are seeing that? And are you properly disposing the `imageStream`?

**Reply by TobiasWolters** [reply]
@dlemstra Thanks for the input and clarification.
I will investigate a bit more and try to come back with a concrete project that can be used to reproduce the issue I'm seeing.

**Comment by TobiasWolters** [comment]
I have not been able to reproduce this issue in an isolated project.
My guess is that the issue I'm seeing is related to a different garbage collector behavior on Linux than on Windows.
As @dlemstra mentioned, this is very unlikely to be related to the pixel cache, so I'll close this discussion for now.
